Nos/Extensions/Font.swift

```diff
@@ -17,6 +17,8 @@ extension Font {
             clarityRegular(textStyle)
         case .medium:
             clarityMedium(textStyle)
+        case .semibold:
```

Comment:
I created this func because the other `static vars` we already have are opaque in their implementation. For instance, check:

```
static var clarityHeadline = Font
        .custom("ClarityCity-SemiBold", size: UIFont.preferredFont(forTextStyle: .headline).pointSize)
```
it doesn't clearly states that the headline is creating is a SemiBold font. In the designs, we usually mix text size and font weight, sometimes is headline semibold, sometimes isn't.

What do you think about getting rid of those declarations, so we avoid using them in the future. Or at least, marking them as deprecated.

Reply:
@martindsq I like this func and would prefer to use it over the others. In fact, I did use this in ProfileEditView.

I was able to delete a couple unused declarations but the rest is outside the scope of this PR. I think we should continue to refactor away usages of things like `clarityHeadline` and move to this func instead. Or I'm fine with having a ticket to remove them all now if you want to create one.

Reply:
Marking them as deprecated should be enough then. SwiftLint will catch any new PR that tries to use those.

Reply:
See https://github.com/planetary-social/nos/pull/947#discussion_r1525451959

and the ticket: https://github.com/planetary-social/nos/issues/955